### PR TITLE
refactor: set validitytime in cache as milliseconds

### PR DESCRIPTION
**Pull request by shojisoma**
Closes https://github.com/SAP/cloud-sdk-backlog/issues/534
Make `Cache` class just accepts milliseconds instead of `DateInputObject` like `new Cache<T>(validityTime?: number)`

### Change
1. Changed `validityTime` to `number`
2. Removed interface `DateInputObject`
3. Adjusted other parts that use Cache class with `DateInputObject`

### CODE REVIEW

File: packages/connectivity/src/scp-cf/cache.ts [hunk 1]
```diff
@@ -48,9 +38,9 @@ export class Cache<T> implements CacheInterface<T> {
    * Default validity period for each entry in cache.
    * If `undefined`, all cached entries will be valid indefinitely.
    */
-  private defaultValidityTime: DateInputObject | undefined;
+  private defaultValidityTime: number | undefined;
```
Comment (florian-richter) [inline packages/connectivity/src/scp-cf/cache.ts:41]:
```suggestion
  private defaultValidityTimeInMs: number | undefined;
```
[req] In my opinion, it is very confusing to just have `number` as a type and not state what unit of measurement is intended

File: packages/connectivity/src/scp-cf/cache.ts [hunk 2]
```diff
@@ -48,9 +38,9 @@ export class Cache<T> implements CacheInterface<T> {
...
 
-  constructor(validityTime?: DateInputObject) {
+  constructor(validityTime?: number) {
```
Comment (florian-richter) [inline packages/connectivity/src/scp-cf/cache.ts:43]:
```suggestion
  constructor(validityTimeInMs?: number) {
```
[req]

File: packages/connectivity/src/scp-cf/cache.ts [hunk 3]
```diff
@@ -104,22 +94,11 @@ function isExpired<T>(item: CacheEntry<T>): boolean {
 }
 
 function inferExpirationTime(
-  expirationTime: DateInputObject | undefined
+  expirationTime: number | undefined
```
Comment (florian-richter) [inline packages/connectivity/src/scp-cf/cache.ts:97]:
```suggestion
  validityTimeInMs: number | undefined
```
[req] Add `InMs` to indicate unit of measurement + Consistent naming, parameter should be called `validityTimeInMs`

File: packages/connectivity/src/scp-cf/cache.ts [hunk 4]
```diff
@@ -104,22 +94,11 @@ function isExpired<T>(item: CacheEntry<T>): boolean {
...
 ): number | undefined {
   return expirationTime
-    ? inferExpirationTimeFromDate(expirationTime)
+    ? new Date()
+        .setMilliseconds(new Date().getMilliseconds() + expirationTime)
+        .valueOf()
```
Comment (florian-richter) [inline packages/connectivity/src/scp-cf/cache.ts:102]: [pp] I like to put `const now = new Date();` above. AFAIK `new` does not bind as strongly as calling a function so you probably intend this: `(new Date()).getMilliseconds()` but actually call this: `new (Date().getMilliseconds())`.

**Review by florian-richter — CHANGES_REQUESTED** [review]

File: packages/connectivity/src/scp-cf/cache.ts (hunk 1 above)
Comment (shojisoma) [reply packages/connectivity/src/scp-cf/cache.ts:41]: thank you for your comment. that's right. I added `InMs`

File: packages/connectivity/src/scp-cf/cache.ts (hunk 2 above)
Comment (shojisoma) [reply packages/connectivity/src/scp-cf/cache.ts:43]: I added `InMs` here too

File: packages/connectivity/src/scp-cf/cache.ts (hunk 3 above)
Comment (shojisoma) [reply packages/connectivity/src/scp-cf/cache.ts:97]: thank you for your comment. I changed the parameter name as your comment for consistent naming.

File: packages/connectivity/src/scp-cf/cache.ts (hunk 4 above)
Comment (shojisoma) [reply packages/connectivity/src/scp-cf/cache.ts:102]: thank you for your comment. I moved out `new Date()` as another const variable.

File: packages/connectivity/src/scp-cf/cache.ts [hunk 5]
```diff
@@ -104,22 +94,12 @@ function isExpired<T>(item: CacheEntry<T>): boolean {
...
 
 function inferExpirationTime(
-  expirationTime: DateInputObject | undefined
+  validityTimeInMs: number | undefined
 ): number | undefined {
-  return expirationTime
-    ? inferExpirationTimeFromDate(expirationTime)
+  const now = new Date();
+  return validityTimeInMs
+    ? new Date()
```
Comment (florian-richter) [inline packages/connectivity/src/scp-cf/cache.ts:101]:
```suggestion
    ? now
```
[req] This also applies here

File: packages/connectivity/src/scp-cf/destination/destination-cache.ts [hunk 6]
```diff
@@ -38,7 +38,7 @@ export interface DestinationCacheInterface {
 export class DefaultDestinationCache implements DestinationCacheInterface {
   cache: Cache<Destination>;
 
-  constructor(validityTime?: DateInputObject) {
+  constructor(validityTime?: number) {
```
Comment (florian-richter) [inline packages/connectivity/src/scp-cf/destination/destination-cache.ts:41]:
```suggestion
  constructor(validityTimeInMs?: number) {
```

**Review by florian-richter — COMMENTED** [review]
A bit more feedback

File: packages/connectivity/src/scp-cf/cache.ts (hunk 5 above)
Comment (shojisoma) [reply packages/connectivity/src/scp-cf/cache.ts:101]: thank you for your review again. I forgot to replace it. I did it.

File: packages/connectivity/src/scp-cf/destination/destination-cache.ts (hunk 6 above)
Comment (shojisoma) [reply packages/connectivity/src/scp-cf/destination/destination-cache.ts:41]: thank you for your review. I should change here also. I add `InMs` here too.

**Review by florian-richter — APPROVED** [review]
👍

**Pull request merged**